tests/components/template/test_alarm_control_panel.py

```diff
@@ -0,0 +1,556 @@
+                    }
+                },
+            }
+        },
+    )
+
+    await hass.async_start()
+    await hass.async_block_till_done()
+
+    assert len(hass.states.async_all()) == 0
+    assert ("invalid template") in caplog.text
+
+
+async def test_invalid_name_does_not_create(hass, caplog):
+    """Test invalid name."""
+    await setup.async_setup_component(
+        hass,
+        "alarm_control_panel",
+        {
+            "alarm_control_panel": {
+                "platform": "template",
+                "panels": {
+                    "bad name here": {
+                        "value_template": "{{ disarmed }}",
+                        "arm_away": {
+                            "service": "alarm_control_panel.alarm_arm_away",
+                            "entity_id": "alarm_control_panel.test",
+                            "data": {"code": "1234"},
+                        },
+                        "arm_home": {
+                            "service": "alarm_control_panel.alarm_arm_home",
+                            "entity_id": "alarm_control_panel.test",
+                            "data": {"code": "1234"},
+                        },
+                        "arm_night": {
+                            "service": "alarm_control_panel.alarm_arm_night",
+                            "entity_id": "alarm_control_panel.test",
+                            "data": {"code": "1234"},
+                        },
+                        "disarm": {
+                            "service": "alarm_control_panel.alarm_disarm",
+                            "entity_id": "alarm_control_panel.test",
+                            "data": {"code": "1234"},
+                        },
+                    }
+                },
+            }
+        },
+    )
+
+    await hass.async_start()
+    await hass.async_block_till_done()
+
+    assert len(hass.states.async_all()) == 0
+    assert ("invalid slug bad name") in caplog.text
+
+
+async def test_invalid_panel_does_not_create(hass, caplog):
+    """Test invalid alarm control panel."""
+    await setup.async_setup_component(
+        hass,
+        "alarm_control_panel",
+        {
+            "alarm_control_panel": {
+                "platform": "template",
+                "wibble": {"test_panel": "Invalid"},
+            }
+        },
+    )
+
+    await hass.async_start()
+    await hass.async_block_till_done()
+
+    assert len(hass.states.async_all()) == 0
+    assert ("[wibble] is an invalid option") in caplog.text
+
+
+async def test_no_panels_does_not_create(hass, caplog):
+    """Test if there are no panels -> no creation."""
+    await setup.async_setup_component(
+        hass, "alarm_control_panel", {"alarm_control_panel": {"platform": "template"}},
+    )
+
+    await hass.async_start()
+    await hass.async_block_till_done()
+
+    assert len(hass.states.async_all()) == 0
+    assert ("required key not provided @ data['panels']") in caplog.text
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4788050)